File: swea/D3_13428.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class D3_13428 {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		
		int T = Integer.parseInt(br.readLine());
		for(int tc = 1; tc <= T; tc++) {
			String board = br.readLine();
			int min, max;
			max = min = Integer.parseInt(board);
			int length = board.length();
			if(length >= 2) { 
				for(int i = 0; i < length; i++) {
					for(int j = i+1; j < length; j++) {
						char[] chars = board.toCharArray();
						chars[i] = board.charAt(j);
						chars[j] = board.charAt(i);
						int result = Integer.parseInt(String.valueOf(chars));
						if(result < Math.pow(10, length-1)) continue;
						
						max = (max < result) ? result : max;
						min = (min > result) ? result : min;
					}
				}
			}
			
			System.out.printf("#%d %d %d\n", tc, min, max);
		}
	}
}

File: swea/D3_13428.cpp
// SWEA 13428 숫자 조작 (D3)_완전 탐색
#include <iostream>
#include <string>
#include <climits>
#include <cmath>
using namespace std;

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int T;	cin >> T;
	for (int tc = 1; tc <= T; tc++) {
		string board;	cin >> board;
		int min = stoi(board), max = stoi(board);
		int size = board.length();
		if (size >= 2) {
			for (int i = 0; i < size; i++) {
				for (int j = i + 1; j < size; j++) {
					string value = board;
					value[i] = board[j];		value[j] = board[i];
					int result = stoi(value);
					if (result < pow(10, (size - 1)))	continue;
					min = (min > result) ? result : min;
					max = (max < result) ? result : max;
				}
			}
		}
		cout << "#" << tc << " ";
		cout << min << " " << max << endl;
	}
}